src/node/http-request.ts

```diff
@@ -0,0 +1,33 @@
+import { HttpRequest, HttpMethod, HttpBody, HttpRequestHeaders } from "../http";
+
+export class HttpRequestImpl implements HttpRequest {
+  constructor(public readonly headers: HttpRequestHeaders,
+    public readonly body: HttpBody,
+    public readonly method: HttpMethod,
+    public readonly url: string) { }
```

Comment:
I would define the parameter orders differently: url, method (default: get), body: (default: ""), headers (default: {})

Reply:
I actually prefer that the default value for those to be provided from the caller. This object life starts on the server implementation(Node or Deno). We know how Node is designed, but we don't have a final idea of how Deno will be designed and this means that in Deno default will have a different meaning. 

I also prefer to keep them required because this constructor is used heavily for mutating the request and the developer might forget to pass inherited values from the previous request.